engine/flink/components-api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala

```diff
@@ -45,16 +45,71 @@ trait FlinkSourceTestSupport[Raw] extends SourceTestSupport[Raw] { self: Source
   * @tparam Raw - type of raw event that is generated by flink source function.
   *             This is needed to handle e.g. syntax suggestions in UI (in sources with explicite @MethodToInvoke).
   */
-trait BasicFlinkSource[Raw] extends FlinkSource with FlinkIntermediateRawSource[Raw] {
+trait BasicFlinkSource[Raw]
+    extends FlinkSource
+    with CustomizableContextInitializerSource[Raw]
+    with CustomizableTimestampWatermarkHandlerSource[Raw]
+    with ExplicitTypeInformationSource[Raw] {
 
   @silent("deprecated")
   def flinkSourceFunction: SourceFunction[Raw]
 
+  override def contextInitializer: ContextInitializer[Raw] = new BasicContextInitializer[Raw](Unknown)
+
   override def sourceStream(
       env: StreamExecutionEnvironment,
       flinkNodeContext: FlinkCustomNodeContext
   ): DataStream[Context] = {
-    prepareSourceStream(env, flinkNodeContext, flinkSourceFunction)
+    FlinkIntermediateRawSourceUtils.prepareSourceStream(
+      env = env,
+      flinkNodeContext = flinkNodeContext,
+      sourceFunction = flinkSourceFunction,
+      typeInformation = typeInformation,
+      timestampWatermarkHandler = timestampAssigner,
+      contextInitializer = contextInitializer
+    )
+  }
+
+}
+
+/**
+ * Source providing default logic for transforming a `DataStreamSource[Raw]` into `DataStream[Context]`. Fulfills
+ * the same role as `BasicFlinkSource`, but is based on `DataStreamSource` instead of `SourceFunction`.
+ * 
+ * @tparam Raw - type of raw event that is generated by flink source function.
+ */
+trait StandardFlinkSource[Raw]
+    extends FlinkSource
+    with CustomizableContextInitializerSource[Raw]
+    with CustomizableTimestampWatermarkHandlerSource[Raw] {
+
+  def initialSourceStream(
+      env: StreamExecutionEnvironment,
+      flinkNodeContext: FlinkCustomNodeContext
+  ): DataStreamSource[Raw]
+
+  override final def sourceStream(
+      env: StreamExecutionEnvironment,
+      flinkNodeContext: FlinkCustomNodeContext
+  ): DataStream[Context] = {
+    FlinkStandardSourceUtils.prepareSource(
```

Comment:
We shouldn't use the class that is not a top level class (in the separate file). How someone would find it?

Reply:
I removed the function and placed the logic directly in the `StandardFlinkSource`. I don't see a reason to reuse this or expose it as a function in our API. I've renamed the file to `StandardFlinkSourceFunctionUtils` because the only thing that's there is a standard transformation from `SourceFunction` into `DataStreamSource`. I've also placed the initalizing function in its own file, since its not related now.